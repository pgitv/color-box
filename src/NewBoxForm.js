import React, { Component } from 'react';

class NewBoxForm extends Component {
  constructor(props) {
    super(props);
    this.state = { width: 0, height: 0, backgroundColor: '' };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleSubmit(evt) {
    evt.preventDefault();
    this.props.createBox({
      width: +this.state.width,
      height: +this.state.height,
      backgroundColor: this.state.backgroundColor
    });
    this.setState({ width: 0, height: 0, backgroundColor: '' });
  }

  handleChange(evt) {
    this.setState({
      [evt.target.name]: evt.target.value
    });
    // [] is ES2016 convention to dynamically assign key value
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <label htmlFor="width">Width:</label>
        <input
          id="width"
          name="width"
          type="number"
          value={this.state.width}
          onChange={this.handleChange}
        />

        <label htmlFor="height">Height:</label>
        <input
          id="height"
          name="height"
          type="number"
          value={this.state.height}
          onChange={this.handleChange}
        />

        <label htmlFor="backgroundColor">Background Color:</label>
        <input
          id="backgroundColor"
          name="backgroundColor"
          value={this.state.backgroundColor}
          onChange={this.handleChange}
        />
        <button>Add a new item!</button>
      </form>
    );
  }
}

export default NewBoxForm;
