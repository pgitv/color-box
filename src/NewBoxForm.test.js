import React from 'react';
import ReactDOM from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NewBoxForm from './NewBoxForm';

// Smoke tests
it('renders without crashing', function() {
  shallow(<NewBoxForm />);
});

// Snapshot tests
it('matches snapshot', function() {
  let wrapper = shallow(<NewBoxForm />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
