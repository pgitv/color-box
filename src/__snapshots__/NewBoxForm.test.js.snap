// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    htmlFor="width"
  >
    Width:
  </label>
  <input
    id="width"
    name="width"
    onChange={[Function]}
    type="number"
    value={0}
  />
  <label
    htmlFor="height"
  >
    Height:
  </label>
  <input
    id="height"
    name="height"
    onChange={[Function]}
    type="number"
    value={0}
  />
  <label
    htmlFor="backgroundColor"
  >
    Background Color:
  </label>
  <input
    id="backgroundColor"
    name="backgroundColor"
    onChange={[Function]}
    value=""
  />
  <button>
    Add a new item!
  </button>
</form>
`;
